dev-qt/qt-creator/qt-creator-8.0.0.ebuild

```diff
@@ -0,0 +1,311 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+LLVM_MAX_SLOT=14
+PLOCALES="cs da de fr hr ja pl ru sl uk zh-CN zh-TW"
+
+inherit cmake llvm optfeature virtualx xdg
+
+DESCRIPTION="Lightweight IDE for C++/QML development centering around Qt"
+HOMEPAGE="https://doc.qt.io/qtcreator/"
+
+if [[ ${PV} == *9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://code.qt.io/${PN}/${PN}.git"
+	EGIT_SUBMODULES=(
+		perfparser
+		qtscript # Need the dev branch
+		src/libs/qlitehtml
+		src/libs/qlitehtml/src/3rdparty/litehtml
+	)
+	# PV=7.9999 -> EGIT_BRANCH=7.0
+	[[ ${PV} =~ [0-9]+\.9999 ]] && EGIT_BRANCH=$(ver_cut 1).0
+	# PV=7.2.9999 -> EGIT_BRANCH=7.2
+	[[ ${PV} =~ [0-9]+\.[0-9]+\.9999 ]] && EGIT_BRANCH=${PV%.9999}
+else
+	MY_PV=${PV/_/-}
+	MY_P=${PN}-opensource-src-${MY_PV}
+	[[ ${MY_PV} == ${PV} ]] && MY_REL=official || MY_REL=development
+	SRC_URI="https://download.qt.io/${MY_REL}_releases/${PN/-}/$(ver_cut 1-2)/${MY_PV}/${MY_P}.tar.xz"
+	S=${WORKDIR}/${MY_P}
+	KEYWORDS="~amd64 ~arm ~x86"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+
+IUSE="+clang debug designer doc imageviewer serialterminal systemd test +qml qt6 wayland webengine"
+
+# TODO: Sort out the web of test deps
+RESTRICT="test"
+
+# minimum Qt version required
+QT5_PV="5.15.2:5="
+QT6_PV="6.2.0:6"
+
+BDEPEND="
+	qt6? (
+		>=dev-qt/qttools-${QT6_PV}[linguist(+)]
+		doc? ( >=dev-qt/qttools-${QT6_PV}[qdoc(+)] )
+	)
+	!qt6? (
+		>=dev-qt/linguist-tools-${QT5_PV}
+		doc? ( >=dev-qt/qdoc-${QT5_PV} )
+	)
+"
+CDEPEND="
+	app-arch/zstd
+	dev-libs/elfutils
```

Comment:
If you mean "Did it depend on the the perfparser tool which was controlled by the perfprofiler flag and is no longer optional in this ebuild iteration?", then yes.  I'll be amending that by going back to the "the way it was in 6.0" approach that you suggested.  If you mean "Doesn't perfparser optionally depend on`app-arch/zstd`?" , then yes,  since 7.0 via automagic.  Which would require introducing yet  another flag, `zstd`, for optional support in a tool which is already already controlled by the `perfprofiler` flag and `sed`ing yet another file, all to support the corner-case of opening a zstd compressed perf data file.  I opted to force a `app-arch/zstd` dependency on the `perfprofiler` flag instead.   perfparser itself, however, depends on `dev-libs/elfutils`.

Reply:
Apologies for the unclear comment. I was referring to elfutils. If it's still only needed by perfparser/perfprofiler then I'd rather keep that optional as a USE flag.

Regarding zstd, I agree that it would be overkill to introduce a separate flag for it. Having it behind `USE=perfprofiler` sounds good to me. That is, assuming it's used only by that plugin, otherwise things get slightly more complicated.